tests/docker/Dockerfile

```diff
@@ -13,10 +13,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# The base image of openjdk:17 is typically oraclelinux:8-slim, which doesn't include apt-get. 
-# Therefore, use openjdk:17-buster instead.
-ARG jdk_version=openjdk:17-buster
-FROM $jdk_version AS build-native-image
+# We need to build the differnet versions of Java in same image, because the
+# zookeeper client 3.4.x can't run under JDK 17, so we need to use JDK 11 in 2.1 ~ 2.3 
+# kafka versions, see KAFKA-17888
+ARG os_version=ubuntu:latest
+FROM $os_version AS build-native-image
+
+ARG jdk_11_version=11.0.2-linux-x64
```

Comment:
exposing `jdk_11_url` is good enough. Additionally, `https://s3-us-west-2.amazonaws.com/kafka-packages` does not offer all distributions so allowing developers to define `jdk_11_version` is meaningless